metadata-ingestion/src/datahub/ingestion/source/schema_inference/object.py

```diff
@@ -0,0 +1,184 @@
+from collections import Counter
+from typing import Any
+from typing import Counter as CounterType
+from typing import Dict, Sequence, Tuple, Union
+
+from mypy_extensions import TypedDict
+
+
+class BasicSchemaDescription(TypedDict):
+    types: CounterType[type]  # field types and times seen
+    count: int  # times the field was seen
+
+
+class SchemaDescription(BasicSchemaDescription):
+    delimited_name: str  # collapsed field name
+    # we use 'mixed' to denote mixed types, so we need a str here
+    type: Union[type, str]  # collapsed type
+    nullable: bool  # if field is ever missing
+
+
+def is_field_nullable(doc: Dict[str, Any], field_path: Tuple) -> bool:
+    """
+    Check if a nested field is nullable in a document from a collection.
+
+    Parameters
+    ----------
+        doc:
+            document to check nullability for
+        field_path:
+            path to nested field to check, ex. ('first_field', 'nested_child', '2nd_nested_child')
+    """
+
+    if not field_path:
+        return True
+
+    field = field_path[0]
+
+    # if field is inside
+    if field in doc:
+
```

Comment:
We can save a lot of screen real-estate by removing lots of empty lines like this.